CHANGELOG.md

```diff
@@ -1,5 +1,8 @@
 # Braintree iOS SDK Release Notes
 
+## 6.15.0 (2024-03-18)
```

Comment:
```suggestion
## unreleased (CI will not run with this filled in)
```

Reply:
This is where CI greps for unreleased: https://github.com/braintree/braintree_ios/blob/8775b31e1e003df7a67bf1428b784eec0637364a/.github/workflows/release.yml#L21-L22. And here is will CI updates the version and date: https://github.com/braintree/braintree_ios/blob/8775b31e1e003df7a67bf1428b784eec0637364a/.github/workflows/release.yml#L38

Reply:
addressed here: c5604acb88ab23f9ff537e5cbc54b4ac565a4bd6